### PR TITLE
Graduate `DownwardAPIHugePages` feature to stable / GA

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This update updates the feature documentation for its GA graduation.

#### Which issue(s) this PR fixes:

Refers to https://github.com/kubernetes/enhancements/issues/2053

#### Special notes for your reviewer:

cc @kubernetes/sig-node-api-reviews 

#### Does this PR introduce a user-facing change?

```release-note
The `DownwardAPIHugePages` kubelet feature graduated to stable / GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
KEP: https://github.com/kubernetes/enhancements/blob/b5b3585/keps/sig-node/2053-downward-api-hugepages/README.md
```

### CODE REVIEW

**Review by mrunalp — APPROVED** [review]

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -897,7 +898,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	DisableKubeletCloudCredentialProviders: {Default: false, PreRelease: featuregate.Alpha},
 
-	DownwardAPIHugePages: {Default: true, PreRelease: featuregate.Beta}, // on by default in 1.22
+	DownwardAPIHugePages: {Default: true, PreRelease: featuregate.GA}, // on by default in 1.22
```
Comment (liggitt) [inline pkg/features/kube_features.go:901]: set `LockToDefault:true`, add a comment to remove in 1.29, and drop code that checks if this gate is enabled

Comment (saschagrunert) [reply pkg/features/kube_features.go]: Yep, done :+1:

**Pull request merged**